web/src/org/purefn/kurosawa/web/server.clj

```diff
@@ -44,26 +45,72 @@
         (log/info "Immutant web server not running")
         this)))
 
-  
+
   ;;----------------------------------------------------------------------------
   log-proto/Logging
   (log-namespaces [_]
     ["org.projectodd.wunderboss.*" "org.xnio.*" "org.xnio" "org.jboss.*"])
-  
+
   (log-configure [this dir]
     (klog/add-component-appender :immutant (log-api/log-namespaces this)
                                  (str dir "/immutant.log"))))
 
 
+(defrecord HttpkitServer
+  [config app server]
+
+  component/Lifecycle
+  (start [this]
+    (if server
+      (do
+        (log/info "HTTP-Kit web server already started")
+        this)
+      (let [_ (log/info "Starting HTTP-Kit web server with" config)
+            app-handle (app/app-handler app)
+            config (-> config
+                       (merge {:error-logger (fn [msg ex] (log/error ex msg))
+                               :warn-logger  (fn [msg ex] (log/warn ex msg))
+                               :event-logger (fn [event-name] (log/info event-name))}))
+            serv-handle (->> {::worker-threads :thread
+                              ::host :ip
+                              ::port :port}
+                             (set/rename-keys config)
+                             seq
+                             flatten
+                             (apply httpkit-server/run-server
+                                    app-handle))]
+        (assoc this :server serv-handle))))
+
+  (stop [this]
+    (if server
+      (do
+        (log/info "Stopping HTTP-Kit web server on port" (::port config))
+        ((:server server))
+        (assoc this :server nil))
+      (do
+        (log/info "HTTP-Kit web server not running")
+        this)))
+
+
+  ;;----------------------------------------------------------------------------
+  log-proto/Logging
```

Comment:
About logging (there is some tech debt here):

Log file treatment is generally hold over from pre-containerized deployments; I don't think we use [init-prod-logging](https://github.com/PureFnOrg/kurosawa/blob/56b6d122c6299e784dd09aa14b1210bd55ff605d/log/src/org/purefn/kurosawa/log/core.clj#L161) anywhere in production.  Rather, [init-dev-logging](https://github.com/PureFnOrg/kurosawa/blob/56b6d122c6299e784dd09aa14b1210bd55ff605d/log/src/org/purefn/kurosawa/log/core.clj#L183) is used and since we pipe all of STDOUT directly to our log aggregator `application name` + `namespace` has been proven effective enough for our workloads in log search.

Given this, `log-namespaces` here is blacklisting all logs from `org.purefn.kurosawa.web.server` from ever reaching STDOUT and therefore aggregation once `init-dev-logging` has been called in the JVM process.  Being unfamiliar with `http-kit` and the signal level from `:error-logger, :event-logger` etc, I'm not sure if that's a desirable outcome!

Reply:
I have omitted the logger namespace exclusion for HTTP-Kit